syntax = "proto3";

package helloworld;

message Empty {
}

message HelloRequest {
	string name = 1;
}

message HelloReply {
	string message = 1;
}

message HelloEvent {
	string message = 1;
}

service Greeter {	
	rpc SayHello (HelloRequest) returns (HelloReply) {}
	rpc Start (Empty) returns (Empty) {}
	rpc Stop (Empty) returns (Empty) {}
	rpc Subscribe (Empty) returns (stream HelloEvent) {}
}
